app/models/procedure.rb

```diff
@@ -963,4 +963,12 @@ def self.human_attribute_name(attribute, options = {})
       super
     end
   end
+
+  def pieces_jointes_list
+    if publiee?
+      published_types_de_champ_public.filter(&:piece_justificative?)
+    else
+      draft_types_de_champ_public.filter(&:piece_justificative?)
+    end
```

Comment:
```suggestion
    active_revision.revision_types_de_champ_public.filter(&:piece_justificative?)
```

Avec ce code, tu embarques mm les pjs qui sont conditionnées et qui ne seront pas forcément demandées. Est-ce voulu ?

Reply:
Alors je n'avais pas cette subtilité en tête - merci !
Du coup si je comprends avec ta suggestion - en fonction des choix dans le formulaire on va peut-être avoir des surprises de PJ ?
Est-ce qu'il ne faudrait pas une liste de PJ "obligatoires" et une liste de PJ "en fonction des situations" - avec un meilleur wording évidemment. Ca serait facile à faire ça ?